bundles/org.eclipse.swt.svg/src/org/eclipse/swt/svg/SVGRasterizer.java

```diff
@@ -0,0 +1,69 @@
+package org.eclipse.swt.svg;
+
+import static java.awt.RenderingHints.*;
+
+import java.awt.*;
+import java.awt.image.*;
+import java.io.*;
+import java.util.*;
+import org.eclipse.swt.graphics.ISVGRasterizer;
+import org.eclipse.swt.graphics.SVGRasterizerRegistry;
+import org.eclipse.swt.graphics.SVGUtil;
+
+import com.github.weisj.jsvg.*;
+import com.github.weisj.jsvg.geometry.size.*;
+import com.github.weisj.jsvg.parser.*;
+
+/**
+ * A rasterizer implementation for converting SVG data into rasterized images.
+ * This class implements the {@code ISVGRasterizer} interface.
+ * 
+ * @since 1.0.0
+ */
+public class SVGRasterizer implements ISVGRasterizer {
+
+	/**
+	 * Initializes the SVG rasterizer by registering an instance of this rasterizer
+	 * with the {@link SVGRasterizerRegistry}.
+	 */
+	public static void intializeSVGRasterizer() {
+		SVGRasterizerRegistry.register(new SVGRasterizer());
+	}
+
+	private final static Map<Key, Object> RENDERING_HINTS = Map.of(KEY_ANTIALIASING, VALUE_ANTIALIAS_ON, //
+			KEY_ALPHA_INTERPOLATION, VALUE_ALPHA_INTERPOLATION_QUALITY, //
+			KEY_COLOR_RENDERING, VALUE_COLOR_RENDER_QUALITY, //
+			KEY_DITHERING, VALUE_DITHER_DISABLE, //
+			KEY_FRACTIONALMETRICS, VALUE_FRACTIONALMETRICS_ON, //
+			KEY_INTERPOLATION, VALUE_INTERPOLATION_BICUBIC, //
+			KEY_RENDERING, VALUE_RENDER_QUALITY, //
+			KEY_STROKE_CONTROL, VALUE_STROKE_PURE, //
+			KEY_TEXT_ANTIALIASING, VALUE_TEXT_ANTIALIAS_ON //
+	);
+
+	@Override
+	public BufferedImage rasterizeSVG(byte[] bytes, float scalingFactor) throws IOException {
+		SVGLoader loader = new SVGLoader();
```

Comment:
If SVGs are only every rasterised from the same thread then this value could be a class member. Initialising it is relatively costly (as it initialises an XML reader internally).

Reply:
Thank you for this hint. It feels like this is a big performance boost.